OData/src/System.Web.OData/OData/Formatter/EdmPrimitiveHelpers.cs

```diff
@@ -84,7 +84,7 @@ public static object ConvertPrimitiveValue(object value, Type type)
                     {
                         DateTimeOffset dateTimeOffsetValue = (DateTimeOffset)value;
                         TimeZoneInfo timeZone = TimeZoneInfoHelper.TimeZone;
-                        dateTimeOffsetValue = dateTimeOffsetValue.ToUniversalTime().ToOffset(timeZone.BaseUtcOffset);
+                        dateTimeOffsetValue = TimeZoneInfo.ConvertTime(dateTimeOffsetValue, timeZone);
```

Comment:
what I do here is try to resolve the daylight saving problem.
This can make sure that I set a -8 timezone, and the payload result has -8:00, not -7:00.